util/src/main/java/tc/oc/pgm/util/nms/NMSHacksNoOp.java

```diff
@@ -0,0 +1,192 @@
+    String strippedKey = key.toLowerCase().replace(" ", "").replace("_", "");
+
+    // Some effects can be parsed by getByName, this only has cases for those that cannot
+    switch (strippedKey) {
+      case "slowness":
+        return PotionEffectType.SLOW;
+      case "haste":
+        return PotionEffectType.FAST_DIGGING;
+      case "miningfatigue":
+        return PotionEffectType.SLOW_DIGGING;
+      case "strength":
+        return PotionEffectType.INCREASE_DAMAGE;
+      case "instanthealth":
+        return PotionEffectType.HEAL;
+      case "instantdamage":
+        return PotionEffectType.HARM;
+      case "jumpboost":
+        return PotionEffectType.JUMP;
+      case "nausea":
+        return PotionEffectType.CONFUSION;
+      case "resistance":
+        return PotionEffectType.DAMAGE_RESISTANCE;
+      default:
+        return PotionEffectType.getByName(key);
+    }
+  }
+
+  @Override
+  public Enchantment getEnchantment(String key) {
+    String strippedKey = key.toLowerCase().replace(" ", "").replace("_", "");
+
+    switch (strippedKey) {
+      case "protection":
+        return Enchantment.PROTECTION_ENVIRONMENTAL;
+      case "fireprotection":
+        return Enchantment.PROTECTION_FIRE;
+      case "featherfalling":
+        return Enchantment.PROTECTION_FALL;
+      case "blastprotection":
+        return Enchantment.PROTECTION_EXPLOSIONS;
+      case "projectileprotection":
+        return Enchantment.PROTECTION_PROJECTILE;
+      case "respiration":
+        return Enchantment.OXYGEN;
+      case "aquaaffinity":
+        return Enchantment.WATER_WORKER;
+      case "thorns":
+        return Enchantment.THORNS;
+      case "depthstrider":
+        return Enchantment.DEPTH_STRIDER;
+      case "sharpness":
+        return Enchantment.DAMAGE_ALL;
+      case "smite":
+        return Enchantment.DAMAGE_UNDEAD;
+      case "baneofarthropods":
+        return Enchantment.DAMAGE_ARTHROPODS;
+      case "knockback":
+        return Enchantment.KNOCKBACK;
+      case "fireaspect":
+        return Enchantment.FIRE_ASPECT;
+      case "looting":
+        return Enchantment.LOOT_BONUS_MOBS;
+      case "efficiency":
+        return Enchantment.DIG_SPEED;
+      case "silktouch":
+        return Enchantment.SILK_TOUCH;
+      case "unbreaking":
+        return Enchantment.DURABILITY;
+      case "fortune":
+        return Enchantment.LOOT_BONUS_BLOCKS;
+      case "power":
+        return Enchantment.ARROW_DAMAGE;
+      case "punch":
+        return Enchantment.ARROW_KNOCKBACK;
+      case "flame":
+        return Enchantment.ARROW_FIRE;
+      case "infinity":
+        return Enchantment.ARROW_INFINITE;
+      case "luckofthesea":
+        return Enchantment.LUCK;
+      case "lure":
+        return Enchantment.LURE;
+      default:
+        return Enchantment.getByName(key);
+    }
+  }
+
+  @Override
+  public long getMonotonicTime(World world) {
+    return Instant.now().toEpochMilli() / 50;
+  }
```

Comment:
i believe this could break some stuff that expects the same time to be returned during the whole tick, as calling this twice within the same tick would return different values. For example see WorldTickClock#now(), usually it'd only create a `new Tick` if the value changed, now it'd create it every time that's called if it's gone over a 50ms border and leaked into the next millisecond.

Reply:
Given that this doesn't seem to change much between versions it could actually be handled fairly well with reflection by my class `util/src/main/java/tc/oc/pgm/util/reflect/MinecraftReflectionUtils.java` from https://github.com/PGMDev/PGM/pull/1199. Given that this code won't actually ever be run in this pull request, could I simply handle it in that PR?

Reply:
I've gone ahead and handled this in the other PR. Let me know if you want it in this PR, although doing so would require more rebasing than I'd like to do for my other PR.

Reply:
I think it'd be nice to merge in this pr, as it can be merged already once this is adressed, rather than waiting and the other PR merging all at once